src/__swaps__/screens/Swap/components/ExchangeRateBubble.tsx

```diff
@@ -1,34 +1,135 @@
+        if (price) {
+          assetToBuyPrice.value = price;
+        }
+      }
 
-  // TODO: This doesn't work when assets change, figure out why...
-  if (!assetToSellLabel.value || !assetToBuyLabel.value) return null;
+      if (current.assetToSell && current.assetToBuy) {
+        runOnJS(SwapInputController.fetchAssetPrices)({
+          assetToSell: current.assetToSell,
+          assetToBuy: current.assetToBuy,
+        });
+      }
+
+      if (current.assetToSellPrice && (!previous?.assetToSellPrice || current.assetToSellPrice !== previous.assetToSellPrice)) {
+        assetToSellPrice.value = current.assetToSellPrice;
+      }
+
+      if (current.assetToBuyPrice && (!previous?.assetToBuyPrice || current.assetToBuyPrice !== previous.assetToBuyPrice)) {
+        assetToBuyPrice.value = current.assetToBuyPrice;
+      }
+
+      if (assetToSellPrice.value && assetToBuyPrice.value) {
+        switch (exchangeRateIndex) {
+          // 1 assetToSell => x assetToBuy
+          case 0: {
+            // TODO: Make the decimal points pretty on the rate conversion
+            const rate = assetToSellPrice.value / assetToBuyPrice.value;
+            fromAssetText.value = `1 ${assetToSellSymbol.value}`;
+            toAssetText.value = `${rate} ${assetToBuySymbol.value}`;
+            break;
+          }
+          // 1 assetToBuy => x assetToSell
+          case 1: {
+            // TODO: Make the decimal points pretty on the rate conversion
+            const rate = assetToBuyPrice.value / assetToSellPrice.value;
+            fromAssetText.value = `1 ${assetToBuySymbol.value}`;
+            toAssetText.value = `${rate} ${assetToSellSymbol.value}`;
+            break;
+          }
+          // assetToSell => native currency
+          case 2: {
+            fromAssetText.value = `1 ${assetToSellSymbol.value}`;
+            toAssetText.value = `$${assetToSellPrice.value.toLocaleString('en-US', {
+              useGrouping: true,
+              minimumFractionDigits: 2,
+              maximumFractionDigits: 2,
+            })}`;
+            break;
+          }
+          // assetToBuy => native currency
+          case 3: {
+            fromAssetText.value = `1 ${assetToBuySymbol.value}`;
+            toAssetText.value = `$${assetToBuyPrice.value.toLocaleString('en-US', {
+              useGrouping: true,
+              minimumFractionDigits: 2,
+              maximumFractionDigits: 2,
+            })}`;
+            break;
+          }
+        }
+      }
+    }
+  );
+
+  const WrapperStyles = useAnimatedStyle(() => {
+    return {
+      borderColor: isDarkMode ? SEPARATOR_COLOR : LIGHT_SEPARATOR_COLOR,
+      borderWidth: THICK_BORDER_WIDTH,
+      opacity: fromAssetText.value && toAssetText.value ? 1 : 0,
+    };
+  });
 
   return (
-    <ButtonPressAnimation scaleTo={0.925} style={{ marginTop: 4 }}>
+    <ButtonPressAnimation onPress={() => setExchangeRateIndex((exchangeRateIndex + 1) % 4)} scaleTo={0.925} style={{ marginTop: 4 }}>
```

Comment:
is 4 # of cases? 🤔

Reply:
yes https://www.figma.com/file/I5NP2G1nT6REuCMCBIZDjR/swaps-2024?type=design&node-id=572-43193&mode=design&t=u6kyqiBx0yvjrozl-0